test/e2e/operator_test.go

```diff
@@ -514,53 +517,63 @@ func TestAWSLoadBalancerControllerWithWAFv2(t *testing.T) {
 		waitForDeletion(t, kubeClient, echoNs, defaultTimeout)
 	}()
 
-	wafClient := wafv2.NewFromConfig(cfg)
-	webACLName := "echoserver-acl"
-	acl, err := findAWSWebACL(wafClient, webACLName)
-	if err != nil {
-		t.Logf("failed to find %q aws wafv2 acl due to %v, continue to creation anyway", webACLName, err)
-	}
-	if acl == nil {
-		t.Logf("AWS WAFv2 ACL %q was not found, creating one", webACLName)
-		createdACL, err := wafClient.CreateWebACL(context.Background(), &wafv2.CreateWebACLInput{
-			DefaultAction: &wafv2types.DefaultAction{Block: &wafv2types.BlockAction{}},
-			Name:          aws.String(webACLName),
-			Scope:         wafv2types.ScopeRegional,
-			VisibilityConfig: &wafv2types.VisibilityConfig{
-				CloudWatchMetricsEnabled: false,
-				MetricName:               aws.String("echoserver"),
-				SampledRequestsEnabled:   false,
-			},
-		})
+	var aclARN string
+	if !isOnROSA() {
+		wafClient := wafv2.NewFromConfig(cfg)
+		webACLName := "echoserver-acl"
+		acl, err := findAWSWebACL(wafClient, webACLName)
 		if err != nil {
-			t.Fatalf("failed to create aws wafv2 acl due to %v", err)
+			t.Logf("failed to find %q aws wafv2 acl due to %v, continue to creation anyway", webACLName, err)
+		}
+		if acl == nil {
+			t.Logf("AWS WAFv2 ACL %q was not found, creating one", webACLName)
+			createdACL, err := wafClient.CreateWebACL(context.Background(), &wafv2.CreateWebACLInput{
+				DefaultAction: &wafv2types.DefaultAction{Block: &wafv2types.BlockAction{}},
+				Name:          aws.String(webACLName),
+				Scope:         wafv2types.ScopeRegional,
+				VisibilityConfig: &wafv2types.VisibilityConfig{
+					CloudWatchMetricsEnabled: false,
+					MetricName:               aws.String("echoserver"),
+					SampledRequestsEnabled:   false,
+				},
+			})
+			if err != nil {
+				t.Fatalf("failed to create aws wafv2 acl due to %v", err)
+			}
+			acl = createdACL.Summary
 		}
-		acl = createdACL.Summary
-	}
 
-	t.Logf("Got AWS WAFv2 ACL. ID: %s, Name: %s", *acl.Id, *acl.Name)
+		aclARN = *acl.ARN
```

Comment:
Can acl ever be nil?

Reply:
No. We try to find it first and if it doesn't exist we try to create it. If the creation fails we fail the test.